spec/requests/api/v1/companies/index_spec.rb

```diff
@@ -0,0 +1,65 @@
+require "rails_helper"
+
+RSpec.describe "Companies Index", type: :request do
+  describe "Get All Companies Endpoint" do
+    let!(:user) { User.create!(name: "Tom", email: "myspace_creator@email.com", password: "test123") }
+    
+    before do
+      post api_v1_sessions_path, params: { email: user.email, password: "test123" }, as: :json
+      @token = JSON.parse(response.body)["token"]
+    end
+
+    it "retrieves all companies but does not share any sensitive data" do
+      Company.create!(user_id: user.id, name: "Google", website: "google.com", street_address: "1600 Amphitheatre Parkway", city: "Mountain View", state: "CA", zip_code: "94043", notes: "Search engine")
+      Company.create!(user_id: user.id, name: "Facebook", website: "facebook.com", street_address: "1 Hacker Way", city: "Menlo Park", state: "CA", zip_code: "94025", notes: "Social media")
+      Company.create!(user_id: user.id, name: "Amazon", website: "amazon.com", street_address: "410 Terry Ave N", city: "Seattle", state: "WA", zip_code: "98109", notes: "E-commerce")
+
+      get "/api/v1/companies", headers: { "Authorization" => "Bearer #{@token}" }, as: :json
+
+      expect(response).to be_successful
+      json = JSON.parse(response.body, symbolize_names: true)
+
+      expect(json[:data].count).to eq(3)
+      expect(json[:data][0][:attributes]).to include(:name, :website, :street_address, :city, :state, :zip_code, :notes)
```

Comment:
using 
``` ruby
json[:data].each do |company| 
expect(company[:attributes]).to include(:name, :website, :street_address, :city, :state, :zip_code, :notes)
``` 
would ensure attribute consistency across company instances instead of checking only the first instance of a company here

Reply:
great suggestion!